.github/workflows/prepare-images.yml

```diff
@@ -2,107 +2,43 @@ name: Prepare images
 
 on:
   workflow_call:
-    outputs:
+    inputs:
       docker_image_tag:
-        description: "linux image for testing"
-        value: ${{ jobs.image-select.outputs.image_tag }}
-      windows_image_tag:
-        description: "windows image for testing"
-        value: ${{ jobs.image-select.outputs.windows_image_tag }}
+        required: true
+        type: string
+      need_linux_image_rebuild:
+        required: true
+        type: boolean
+      need_windows_vcpkg_rebuild:
+        required: true
+        type: boolean
+      vcpkg_version:
+        required: true
+        type: string
 
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
-env:
-  VCPKG-VERSION: '2024.10.21'
-
 jobs:
   image-select:
```

Comment:
mb we can join `image-select` and `linux-image-build-upload` as they both handle only linux docker images?